### PR TITLE
nixUnstable: add util-linuxMinimal to nativeBuildInputs; skip flakey test

### PR DESCRIPTION
**Pull request by cole-h**
Fixes an issue in tests where they can fail with `../common.sh: line 92:
unshare: command not found`.

###### Motivation for this change

Similar to https://github.com/NixOS/nixpkgs/pull/114602, but is more in-line with what happens upstream: uses `util-linuxMinimal` rather than `util-linux`, and places it in `buildInputs` rather than `nativeBuildInputs`.

EDIT: Upstream is trending towards moving it into nativeBuildInputs, and we want to see if this unblocks channels, so I moved it to nativeBuildInputs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by sternenseemann — APPROVED** [review]
LGTM, builds for me locally.

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -57,6 +57,7 @@ common =
         ++ lib.optional (stdenv.isLinux || stdenv.isDarwin) libsodium
         ++ lib.optionals is24 [ libarchive gmock lowdown ]
         ++ lib.optional (is24 && stdenv.isx86_64) libcpuid
+        ++ lib.optionals (is24 && stdenv.isLinux) [ util-linuxMinimal ]
```
Comment (sternenseemann) [inline pkgs/tools/package-management/nix/default.nix:60]: If you feel like rebasing again, that would be a case for `lib.optional` :)

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117295` at e9eab755 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>hydra-unstable</li>
    <li>nix-direnv</li>
    <li>nix-update</li>
    <li>nixFlakes (nixUnstable)</li>
    <li>nixpkgs-review</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/package-management/nix/default.nix:205:16:
```
    |
205 |     patches = [(
    |                ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/package-management/nix/default.nix:68:17:
```
   |
68 |                 ./aws-sdk-cpp-TransferManager-ContentEncoding.patch
   |                 ^
```
</li>
  </ul>
</details>

File: pkgs/tools/package-management/nix/default.nix (hunk 1 above)
Comment (cole-h) [reply pkgs/tools/package-management/nix/default.nix:60]: This was actually intentional: https://github.com/NixOS/nixpkgs/pull/116274#discussion_r593834474

Comment (sternenseemann) [reply pkgs/tools/package-management/nix/default.nix:60]: Fair enough.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 117295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>5 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>hydra-unstable: log was empty</li>
<li>nix-direnv: log was empty</li>
<li>nix-update: log was empty</li>
<li>nixFlakes: <a href="https://termbin.com/ub6ax">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/b55087c30fd41237316d8447804f9013/raw/index.html">pretty log </a></li>
<li>nixpkgs-review: log was empty</li>
  </ul>
</details>
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>(nixUnstable): log was empty</li>
  </ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
hydra-unstable:
<details>


warning: unclear-gpl
`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/tools/misc/hydra/common.nix:133:5:

```
    |
133 |     license = licenses.gpl3;
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: build-tools-in-build-inputs
autoconf is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: build-tools-in-build-inputs
automake is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: build-tools-in-build-inputs
libtool is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: build-tools-in-build-inputs
makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: build-tools-in-build-inputs
subversion is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: build-tools-in-build-inputs
unzip is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: unused-argument
Unused argument: `darcs`.
Near pkgs/development/tools/misc/hydra/common.nix:3:23:

```
  |
3 | , top-git, mercurial, darcs, subversion, breezy, openssl, bzip2, libxslt
  |                       ^
```
</details>
nix-direnv:
<details>


warning: unused-argument
Unused argument: `nix`.
Near pkgs/tools/misc/nix-direnv/default.nix:1:42:

```
  |
1 | { lib, stdenv, fetchFromGitHub, gnugrep, nix, nixFlakes }:
  |                                          ^
```
</details>
</details>

**Comment by cole-h** [comment]
Waiting on the outcome of https://github.com/NixOS/nix/pull/4659.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 117295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>hydra-unstable</li>
<li>nix-direnv</li>
<li>nix-update</li>
<li>nixFlakes (nixUnstable)</li>
<li>nixpkgs-review</li>
</ul>
</details>

**Comment by cole-h** [comment]
We want to unblock channels, so I'm going to merge as soon as this is green. If this is actually blocking, then we want this in ASAP to prepare for upcoming security bumps. If it isn't, we'll need to disable the flakey test until upstream can figure out why this is happening.

**Review by infinisil — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 117295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>hydra-unstable</li>
<li>nix-direnv</li>
<li>nix-review (nixpkgs-review)</li>
<li>nix-update</li>
<li>nixFlakes (nixUnstable)</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 117295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>hydra-unstable</li>
<li>nix-direnv</li>
<li>nix-review (nixpkgs-review)</li>
<li>nix-update</li>
<li>nixFlakes (nixUnstable)</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 117295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>hydra-unstable</li>
<li>nix-direnv</li>
<li>nix-review (nixpkgs-review)</li>
<li>nix-update</li>
<li>nixFlakes (nixUnstable)</li>
</ul>
</details>

**Pull request merged**